grc20-core/src/mapping/entity/find_path.rs

```diff
@@ -0,0 +1,162 @@
+    neo4j: neo4rs::Graph,
+    id1: String,
+    id2: String,
+    filter: EntityFilter,
+    order_by: Option<FieldOrderBy>,
+    limit: usize,
+    skip: Option<usize>,
+    space_id: Option<PropFilter<String>>,
+    version: VersionFilter,
+    _phantom: std::marker::PhantomData<T>,
+}
+
+impl<T> FindPathQuery<T> {
+    pub(super) fn new(neo4j: &neo4rs::Graph, id1: String, id2: String) -> Self {
+        Self {
+            neo4j: neo4j.clone(),
+            id1,
+            id2,
+            filter: EntityFilter::default(),
+            order_by: None,
+            limit: 100,
+            skip: None,
+            space_id: None,
+            version: VersionFilter::default(),
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    pub fn id(mut self, id: PropFilter<String>) -> Self {
+        self.filter.id = Some(id);
+        self
+    }
+
+    pub fn attribute(mut self, attribute: AttributeFilter) -> Self {
+        self.filter.attributes.push(attribute);
+        self
+    }
+
+    pub fn attribute_mut(&mut self, attribute: AttributeFilter) {
+        self.filter.attributes.push(attribute);
+    }
+
+    pub fn attributes(mut self, attributes: impl IntoIterator<Item = AttributeFilter>) -> Self {
+        self.filter.attributes.extend(attributes);
+        self
+    }
+
+    pub fn attributes_mut(&mut self, attributes: impl IntoIterator<Item = AttributeFilter>) {
+        self.filter.attributes.extend(attributes);
+    }
+
+    pub fn limit(mut self, limit: usize) -> Self {
+        self.limit = limit;
+        self
+    }
+
+    pub fn skip(mut self, skip: usize) -> Self {
+        self.skip = Some(skip);
+        self
+    }
+
+    /// Overwrite the current filter with a new one
+    pub fn with_filter(mut self, filter: EntityFilter) -> Self {
+        self.filter = filter;
+        self
+    }
+
+    pub fn order_by(mut self, order_by: FieldOrderBy) -> Self {
+        self.order_by = Some(order_by);
+        self
+    }
+
+    pub fn order_by_mut(&mut self, order_by: FieldOrderBy) {
+        self.order_by = Some(order_by);
+    }
+
+    pub fn space_id(mut self, space_id: impl Into<PropFilter<String>>) -> Self {
+        self.space_id = Some(space_id.into());
+        self
+    }
+
+    pub fn version(mut self, space_version: String) -> Self {
+        self.version.version_mut(space_version);
+        self
+    }
+
+    pub fn version_opt(mut self, space_version: Option<String>) -> Self {
+        self.version.version_opt(space_version);
+        self
+    }
+
+    fn subquery(&self) -> QueryBuilder {
+        QueryBuilder::default()
+            .subquery(MatchQuery::new(
+                "p = allShortestPaths((e1:Entity {id: $id1}) -[:RELATION*1..5]-(e2:Entity {id: $id2}))",
+            ))
+            .limit(self.limit)
+            .params("id1", self.id1.clone())
+            .params("id2", self.id2.clone())
+    }
+}
+
+impl<T: FromAttributes> Query<Vec<Relation>> for FindPathQuery<T> {
```

Comment:
In queries that return data, the generic type `T` is used to indicate what shape of data will be returned. However, since here were returning a "hardcoded" `Vec<Relation>` as the data, we don't actually need `T`.

There are two possible fixes:
1. (recommended) Change the `impl` to `impl Query<Vec<Relation>> for FindPathQuery<Vec<Relation>>`
2. Remove `T` from `FindPathQuery` (not ideal since we might want to support more return types in the future)